mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -657,14 +662,19 @@ def _hard_delete_run(self, run_id):
             run = self._get_run(run_uuid=run_id, session=session)
             session.delete(run)
 
-    def _get_deleted_runs(self):
+    def _get_deleted_runs(self, older_than=0):
+        """
+        Get all deleted run ids.
+        Args:
+            older_than: get runs that is older than this variable in number of milliseconds.
+                        defaults to 0 ms to get all deleted runs.
+        """
+        current_time = int(time.time() * 1000)
         with self.ManagedSessionMaker() as session:
-            run_ids = (
-                session.query(SqlRun.run_uuid)
-                .filter(SqlRun.lifecycle_stage == LifecycleStage.DELETED)
-                .all()
+            runs = (
+                session.query(SqlRun).filter(SqlRun.lifecycle_stage == LifecycleStage.DELETED).all()
```

Comment:
Can we filter by `deleted_time` when fetching runs from DB?

```suggestion
            runs = (
                session.query(SqlRun)
                .filter(
                    SqlRun.lifecycle_stage == LifecycleStage.DELETED,
                    SqlRun.deleted_time < (current_time - older_than),
                )
                .all()
```

Reply:
👌